superset/views/core.py

```diff
@@ -2591,6 +2600,15 @@ def sql_json_exec(
         # Set tmp_table_name for CTA
         if select_as_cta and mydb.force_ctas_schema:
             tmp_table_name = f"{mydb.force_ctas_schema}.{tmp_table_name}"
+        elif select_as_cta:
+            dest_schema_name = get_cta_schema_name(
+                schema, sql, g.user.username if g.user else None
```

Comment:
`g.user` will only be there if you're in sync mode (inside the scope of a web request), won't work on Celery.

Reply:
`views/core.py` is in the scope of the web request, this happens before query object is created and passed to the celery worker - we should be safe here.